From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: UN1CA <noreply@unica.local>
Date: Thu, 2 Jul 2026 00:00:00 +0000
Subject: [PATCH] Force Audio Eraser global parameter apply

SystemUI sends package-aware AUDIO_ERASER_EFFECT values to SoundAlive's
AudioEraserProvider for YouTube and other locally allowed media apps. The
provider writes the requested mode for all audio paths, but skips the direct
EffectManager.setEffectVoiceBoostAll() call on some Samsung audio routes and
depends on a SoundAlive local listener instead.

Force the direct global parameter apply for both the new AudioEraserProvider
path and the legacy SAContentProvider path so non-Gallery playback gets the
same g_voice_booster_preset_all update immediately.
---
 .../soundalive/compatibility/AudioEraserProvider.smali        | 2 --
 .../sec/android/app/soundalive/compatibility/SAContentProvider.smali | 2 --
 2 files changed, 4 deletions(-)

diff --git a/smali/com/sec/android/app/soundalive/compatibility/AudioEraserProvider.smali b/smali/com/sec/android/app/soundalive/compatibility/AudioEraserProvider.smali
index 1111111..2222222 100644
--- a/smali/com/sec/android/app/soundalive/compatibility/AudioEraserProvider.smali
+++ b/smali/com/sec/android/app/soundalive/compatibility/AudioEraserProvider.smali
@@ -293,8 +293,6 @@
 
     move-result p0
 
-    if-nez p0, :cond_b
-
     sget-object p0, Lt5/g;->a:Lt5/h;
 
     invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
diff --git a/smali/com/sec/android/app/soundalive/compatibility/SAContentProvider.smali b/smali/com/sec/android/app/soundalive/compatibility/SAContentProvider.smali
index 1111111..2222222 100644
--- a/smali/com/sec/android/app/soundalive/compatibility/SAContentProvider.smali
+++ b/smali/com/sec/android/app/soundalive/compatibility/SAContentProvider.smali
@@ -2395,8 +2395,6 @@
 
     move-result v0
 
-    if-nez v0, :cond_3d
-
     sget-object v0, Lt5/g;->a:Lt5/h;
 
     invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
-- 
2.43.0
